contribs/github-bot/comment.go

```diff
@@ -0,0 +1,314 @@
+	if !ok {
+		return errors.New("unable to get comment user on issue comment event")
+	}
+
+	login, ok := author["login"].(string)
+	if !ok {
+		return errors.New("unable to get comment user login on issue comment event")
+	}
+
+	// If comment edition author is not the bot, return
+	if login != authUser.GetLogin() {
+		return nil
+	}
+
+	// Get comment current body
+	current, ok := comment["body"].(string)
+	if !ok {
+		return errors.New("unable to get comment body on issue comment event")
+	}
+
+	// Get comment updated body
+	changes, ok := actionCtx.Event["changes"].(map[string]any)
+	if !ok {
+		return errors.New("unable to get changes on issue comment event")
+	}
+
+	changesBody, ok := changes["body"].(map[string]any)
+	if !ok {
+		return errors.New("unable to get changes body on issue comment event")
+	}
+
+	previous, ok := changesBody["from"].(string)
+	if !ok {
+		return errors.New("unable to get changes body content on issue comment event")
+	}
+
+	// Get PR number from GitHub Actions context
+	issue, ok := actionCtx.Event["issue"].(map[string]any)
+	if !ok {
+		return errors.New("unable to get issue on issue comment event")
+	}
+
+	num, ok := issue["number"].(float64)
+	if !ok || num <= 0 {
+		return errors.New("unable to get issue number on issue comment event")
+	}
+
+	// Check if change is only a checkbox being checked or unckecked
+	if checkboxes.ReplaceAllString(current, "") != checkboxes.ReplaceAllString(previous, "") {
+		// If not, restore previous comment body
+		if !gh.DryRun {
+			gh.SetBotComment(previous, int(num))
+		}
+		return errors.New("bot comment edited outside of checkboxes")
+	}
+
+	// Check if actor / comment editor has permission to modify changed boxes
+	currentChecks := getCommentManualChecks(current)
+	previousChecks := getCommentManualChecks(previous)
+	edited := ""
+	for key := range currentChecks {
+		// If there is no diff for this check, ignore it
+		if currentChecks[key][0] == previousChecks[key][0] {
+			continue
+		}
+
+		// Get teams allowed to edit this box from config
+		var teams []string
+		found := false
+		_, manualRules := config(gh)
+
+		for _, manualRule := range manualRules {
+			if manualRule.description == key {
+				found = true
+				teams = manualRule.teams
+			}
+		}
+
+		// If rule were not found, return to reprocess the bot comment entirely
+		// (maybe bot config was updated since last run?)
+		if !found {
+			gh.Logger.Debugf("Updated rule not found in config: %s", key)
+			return nil
+		}
+
+		// If teams specified in rule, check if actor is a member of one of them
+		if len(teams) > 0 {
+			if gh.IsUserInTeams(actionCtx.Actor, teams) {
+				if !gh.DryRun {
+					gh.SetBotComment(previous, int(num))
+				}
+				return errors.New("checkbox edited by a user not allowed to")
+			}
+		}
+
+		// This regex capture only the line of the current check
+		specificManualCheck := regexp.MustCompile(fmt.Sprintf(`(?m:^- \[%s\] %s.*$)`, currentChecks[key][0], key))
```

Comment:
Could this not be a simple string instead?

Reply:
It was what I did initially, but given the brittle nature of the system, it seemed safer to use a regex to match the entire line from the beginning `^` to the end `$` (with the optional "checked by", etc.) to ensure that the whole line is overwritten with a correct value each time, and to avoid issues in corner cases where the comment might be edited multiple times within a few seconds without the bot having time to catch up, resulting in a weird state for the comment diff.

Reply:
Okay; seems kind of overkill but whatever. Can you `regexp.QuoteMeta` the key?

Reply:
Done